gapis/resolve/events.go

```diff
@@ -43,13 +43,24 @@ func Events(ctx context.Context, p *path.Events) (*service.Events, error) {
 	events := []*service.Event{}
 
 	s := c.NewState()
+	lastCmd := api.CmdID(0)
+	delayedFirstInFrame := false
```

Comment:
This took me a while to understand.

How about:

```go
var pending []*serviceEvent
api.ForeachCmd(ctx, c.Commands, func(ctx context.Context, id api.CmdID, cmd api.Cmd) error {
	cmd.Mutate(ctx, id, s, nil)

	// TODO: Add event generation to the API files.
	if !filter(id, cmd, s) {
		return nil
	}

	events, pending = append(events, pending...), nil
```
...
```go
	if p.FirstInFrame && f.IsEndOfFrame() {
		pending = append(pending, &service.Event{true
			Kind:    service.EventKind_FirstInFrame,		
			Command: p.Capture.Command(uint64(id) + 1),		
		})
	}
```

Reply:
Done.